rusk-abi/src/dusk.rs

```diff
@@ -0,0 +1,48 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Dusk denomination.
+
+const DUSK_UNIT: f64 = 1_000_000_000.0;
+
+/// The minimum increment of Dusk.
+pub const LUX: Dusk = dusk(1.0 / DUSK_UNIT);
+
+/// The Dusk denomination. Use the [`dusk`] function to convert from floating
+/// point format, and the [`FromDusk`] trait to convert back.
+///
+/// Values of Dusk should *never* be set directly. Instead they should always
+/// use a call to the [`dusk`] function. If increments of the smallest
+/// denomination are desired, the [`LUX`] constant can be used.
+pub type Dusk = u64;
+
+/// Convert to Dusk from floating point format.
```

Comment:
It should be `Converts`. Also I would rephrase as `Converts from floating point format to Dusk`, the formula `from…to` it's usually preferred, and it will be consistent with the comment below.